src/PackagePrivate/InMemoryTermIdsAcquirer.php

```diff
@@ -46,4 +46,19 @@ public function cleanTerms( array $termInLangIds ) {
 		}
 	}
 
+	public function hasTerms() {
+		$empty = true;
+		// if there's any leaf element in terms then it's not empty, otherwise we consider
+		// the terms array empty even if it had some sub-arrays that are also empty by
+		// this definition
+		array_walk_recursive(
```

Comment:
This is going to walk the whole array all the time :/ But given this is probably only going to be used in tests, I guess that's bearable.

Reply:
yeah this array isn't expected to contain more than tens of elements for testing purposes (hopefully).

Reply:
Is the old version that used assertEmpty on the terms field broken? This looks like a behavior change to me. I already asked about it on the PR that motivated this change and got no answer.

Reply:
I might have got lost with the conversation and misunderstood it (though I doubt).

You may ignore this PR entirely or just revert my changes (I can do it on command ;) if you think it is completely irrelevant to the original improvement/discussion. Sorry to have wasted your time in that case :)

Reply:
The first version did

    assertEmpty( $terms )

Then I changed it do

    assertTrue( empty( $terms ) )

AFAIK that is equivalent. But Lucas said I broke it. Your change here suggests you think it was indeed broken. Where did it get broken?

Reply:
Let's not sweat this one anymore.. it is so trivial anyway. Either merge or kill at will please